src/jit/emitxarch.cpp

```diff
@@ -2847,15 +2847,34 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
     assert(!dst->isContained() ||
            dst->isContainedIndir() ||
            dst->isContainedLclField() ||
+           dst->isContainedLclVar() ||  // this is essentially a contained LclField with zero offset
```

Comment:
This is a very confusing comment. I would never think of a ContainedLclVar as a ContainedLclField with a zero offset. Maybe this comment is here because this is newly added, but in future I think the comment would be more confusing than helpful. Instead, I would think of a ContainedLclField as a ContainedLclVar with a non-zero offset.